pkg/image/admission/imagequalify/qualify_test.go

```diff
@@ -0,0 +1,317 @@
+package imagequalify_test
+
+import (
+	"bytes"
+	"regexp"
+	"runtime"
+	"testing"
+
+	configapilatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	"github.com/openshift/origin/pkg/image/admission/imagequalify"
+	"github.com/openshift/origin/pkg/image/admission/imagequalify/api"
+)
+
+type testcase struct {
+	image    string
+	expected string
+}
+
+// Match Test<XXX> func name in a package specifier.
+//
+// For example, it will match TestFoo in:
+//   github.io/xxx/foo_test.TestFoo
+var testNameRegexp = regexp.MustCompile(`\.(Test[\p{L}_\p{N}]+)$`)
+
+func testName() string {
+	pc := make([]uintptr, 32)
+	n := runtime.Callers(0, pc)
+
+	for i := 0; i < n; i++ {
+		fn := runtime.FuncForPC(pc[i])
+		matches := testNameRegexp.FindStringSubmatch(fn.Name())
+		if matches == nil {
+			continue
+		}
+		return matches[1]
+	}
+
+	panic("test name could not be discovered; try increasing stack depth?")
+}
+
+func parseQualifyRules(rules []api.ImageQualifyRule) (*api.ImageQualifyConfig, error) {
+	config, err := configapilatest.WriteYAML(&api.ImageQualifyConfig{
+		Rules: rules,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return imagequalify.ReadConfig(bytes.NewReader(config))
+}
+
+func testQualify(t *testing.T, rules []api.ImageQualifyRule, tests []testcase) {
```

Comment:
call `t.Helper()` instead of `testName()`... makes t.Fatalf print the line and file of the caller

see https://golang.org/pkg/testing/#T.Helper

Reply:
Ah, thank you! Being carrying that wart for a little while now.